### PR TITLE
New facility integration

### PR DESCRIPTION
**Pull request by izgeorge**
Integrates adding a new facility with the new fields in the database. Should display the confirmation screen, then on next, route to the signup with the ID of the newly created facility

### CODE REVIEW

File: src/components/InputDropdown/index.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ class InputDropdown extends React.Component {
   }
 
   componentWillUnmount() {
-    window.removeEventListener('mousedown')
+    window.removeEventListener('mousedown', () => false)
```
Comment (joshrounsville) [inline src/components/InputDropdown/index.js]: I added a fix for this that might be closer to what was intended: https://github.com/helpsupply/helpsupply/pull/81/files/2b02fac50ca47940c2a0e4d6f86fe3b1fa5729d7#diff-ab45b32c1a60f1bea17b342d7592bc0c

What do you think?

File: src/components/Form/index.js [hunk 2]
```diff
@@ -16,22 +16,28 @@ class Form extends React.Component {
     disabled: PropTypes.bool,
     onSubmit: PropTypes.func.isRequired,
     title: PropTypes.string,
-  };
+  }
 
   constructor(props) {
-    super(props);
+    super(props)
+    this.onSubmit = this.onSubmit.bind(this)
```
Comment (joshrounsville) [inline src/components/Form/index.js]: Seems like we use arrow functions to handle this binding in a number of places. Could we do the same here and avoid the `constructor` all together?

File: src/containers/FacilityForm.js [hunk 3]
```diff
@@ -7,47 +7,89 @@ import InputDropdown from 'components/InputDropdown'
...
+        dropSiteFacilityName: '',
+        dropSiteZip: '',
+        dropSiteAddress: '',
+        dropSiteCity: '',
+        dropSiteState: '',
+        dropSiteUrl: '',
+      },
     }
     this.handleFieldChange = this.handleFieldChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
```
Comment (joshrounsville) [inline src/containers/FacilityForm.js]: Same question re: binding and removing the constructor.

File: src/components/InputDropdown/index.js (hunk 1 above)
Comment (izgeorge) [reply src/components/InputDropdown/index.js]: thank you!

File: src/components/Form/index.js [hunk 4]
```diff
@@ -1,52 +1,49 @@
...
-    buttonLabel: PropTypes.string,
-    description: PropTypes.string,
-    disabled: PropTypes.bool,
-    onSubmit: PropTypes.func.isRequired,
-    title: PropTypes.string,
-  };
-
-  constructor(props) {
-    super(props);
+export const Form = ({
```
Comment (joshrounsville) [inline src/components/Form/index.js]: 🙌

File: src/containers/DropSiteForm.js [hunk 5]
```diff
@@ -1,93 +1,86 @@
...
+  const [fields, setFields] = useState({
+    dropSiteId: dropSite.location_id,
+    dropSiteDescription: '',
+    dropSiteAddress: dropSite.dropSiteAddress || '',
+    dropSiteRequirements: '',
+    dropSiteName: '',
+    dropSitePhone: '',
+    dropSiteNotes: '',
+    requestWillingToPay: false,
+  })
```
Comment (joshrounsville) [inline src/containers/DropSiteForm.js]: Nice!

**Pull request merged**